packages/ui-text-input/src/TextInput/styles.ts

```diff
@@ -198,30 +186,21 @@ const generateStyle = (
     },
     layout: {
       label: 'textInput__layout',
-      ...flexBase,
-      ...(!shouldNotWrap && { flexWrap: 'wrap' })
-    },
-    beforeElement: {
-      display: 'inline-flex',
+      ...viewBase,
+      display: 'flex',
+      flexGrow: 1,
       alignItems: 'center',
-      label: 'textInput__beforeElement',
-      ...flexItemBase,
-      paddingInlineStart: componentTheme.padding,
-      // we only override the padding once the width is calculated,
-      // it needs the padding on render
-      ...(beforeElementHasWidth === false && {
-        paddingInlineStart: 0
-      })
-    },
-    innerWrapper: {
-      label: 'textInput__innerWrapper',
-      ...flexItemBase,
-      minWidth: '0.0625rem',
-      flexShrink: 1,
-      flexGrow: 1
+      justifyContent: 'flex-start',
+      flexDirection: 'row',
+      // Apply the style if the first child has NOT the inputLayout class
+      '&:has(> :first-child:not([class$="textInput__inputLayout"]))': {
+        paddingInlineStart: componentTheme.padding
+      },
```

Comment:
This method is slightly worse than the previous one, but does not need JS (now it applies padding if someone puts in a 0 width `beforeElement`)